pkg/providers/cloudstack/decoder/decoder.go

```diff
@@ -16,10 +17,53 @@ const (
 	EksaCloudStackHostPathToMount         = "EKSA_CLOUDSTACK_HOST_PATHS_TO_MOUNT"
 	defaultVerifySslValue                 = "true"
 	CloudStackGlobalAZ                    = "global"
+
+	APIKeyKey    = "api-key"
+	SecretKeyKey = "secret-key"
+	APIUrlKey    = "api-url"
+	VerifySslKey = "verify-ssl"
 )
 
-// ParseCloudStackSecret parses the input b64 string into the ini object to extract out the api key, secret key, and url.
-func ParseCloudStackSecret() (*CloudStackExecConfig, error) {
+// ParseCloudStackCredsFromSecrets parses a list of secrets to extract out the api keys, secret keys, and urls.
+func ParseCloudStackCredsFromSecrets(secrets []apiv1.Secret) (*CloudStackExecConfig, error) {
+	var cloudstackProfiles []CloudStackProfileConfig
+	for _, secret := range secrets {
+		apiKey, ok := secret.Data[APIKeyKey]
+		if !ok {
+			return nil, fmt.Errorf("secret %s is missing required key %s", secret.Name, APIKeyKey)
+		}
+		secretKey, ok := secret.Data[SecretKeyKey]
+		if !ok {
+			return nil, fmt.Errorf("secret %s is missing required key %s", secret.Name, SecretKeyKey)
+		}
+		apiURL, ok := secret.Data[APIUrlKey]
+		if !ok {
+			return nil, fmt.Errorf("secret %s is missing required key %s", secret.Name, APIUrlKey)
+		}
+		verifySsl, ok := secret.Data[VerifySslKey]
+		if !ok {
+			verifySsl = []byte(defaultVerifySslValue)
+		}
+		cloudstackProfiles = append(cloudstackProfiles, CloudStackProfileConfig{
+			Name:          secret.Name,
+			ApiKey:        string(apiKey),
+			SecretKey:     string(secretKey),
+			ManagementUrl: string(apiURL),
+			VerifySsl:     string(verifySsl),
+		})
+	}
+
+	if len(cloudstackProfiles) == 0 {
```

Comment:
could make this check at the beginning by checking the secret slice len?

Reply:
Sure